File: server/datasets/java_source-code/Multiple-G1/renamed_41_bst.java
class Node_5 {
    int val_5;
    Node_5 lhs_5, rhs_5;
    public Node_5(int item) {
        val_5 = item;
        lhs_5 = rhs_5 = null;
    }
}
public class Tree_5 {
    public static void main(String[] args) {
        Tree_5 bst = new Tree_5();
        bst.ins_5(50);
        bst.srch_5(50);
    }
    Node_5 rt_5;
    public void ins_5(int val_5) {
        rt_5 = in_rec_5(rt_5, val_5);
    }
    Node_5 in_rec_5(Node_5 rt_5, int val_5) {
        if (rt_5 == null) return new Node_5(val_5);
        if (val_5 < rt_5.val_5) rt_5.lhs_5 = in_rec_5(rt_5.lhs_5, val_5);
        else if (val_5 > rt_5.val_5) rt_5.rhs_5 = in_rec_5(rt_5.rhs_5, val_5);
        return rt_5;
    }
    Node_5 srch_5(int val_5) {
        return sr_rec_5(rt_5, val_5);
    }
    Node_5 sr_rec_5(Node_5 rt_5, int val_5) {
        if (rt_5 == null || rt_5.val_5 == val_5) return rt_5;
        if (rt_5.val_5 < val_5) return sr_rec_5(rt_5.rhs_5, val_5);
        return sr_rec_5(rt_5.lhs_5, val_5);
    }
}

File: server/datasets/python_source-code/Multiple-G1/renamed_41_bst.py
class Node_5:
    def __init__(self, val_5):
        self.val_5 = val_5
        self.lhs_5 = None
        self.rhs_5 = None

class Tree_5:
    def __init__(self):
        self.rt_5 = None

    def ins_5(self, val_5):
        if self.rt_5 is None:
            self.rt_5 = Node_5(val_5)
        else:
            self._ins_5_rec(self.rt_5, val_5)

    def _ins_5_rec(self, nd_5, val_5):
        if val_5 < nd_5.val_5:
            if nd_5.lhs_5 is None:
                nd_5.lhs_5 = Node_5(val_5)
            else:
                self._ins_5_rec(nd_5.lhs_5, val_5)
        elif val_5 > nd_5.val_5:
            if nd_5.rhs_5 is None:
                nd_5.rhs_5 = Node_5(val_5)
            else:
                self._ins_5_rec(nd_5.rhs_5, val_5)

    def srch_5(self, val_5):
        return self._srch_5_rec(self.rt_5, val_5)

    def _srch_5_rec(self, nd_5, val_5):
        if nd_5 is None or nd_5.val_5 == val_5:
            return nd_5
        if val_5 < nd_5.val_5:
            return self._srch_5_rec(nd_5.lhs_5, val_5)
        return self._srch_5_rec(nd_5.rhs_5, val_5)
